hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/avro/AvroMessage.java

```diff
@@ -14,17 +16,20 @@ public class AvroMessage implements Message {
     private final long timestamp;
     private final CompiledSchema<Schema> schema;
     private final String partitionKey;
+    private final Map<String, String> extraRequestHeaders;
```

Comment:
Probably it will not be needed if you apply my comment about the backup storage.